src/main/java/duke/utils/TaskList.java

```diff
@@ -0,0 +1,107 @@
+package duke.utils;
+
+import duke.DukeException;
+import duke.task.Task;
+
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+/**
+ * Represents the list of tasks that Duke keeps track of.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Adds a task to the task list.
+     *
+     * @param task The new task to be added.
+     */
+    public void addTask(Task task) {
+        tasks.add(task);
+
+        System.out.println("\n  _______________");
+        System.out.println("  Added: " + task);
+        System.out.println(String.format("  Now you have %d tasks in the list", tasks.size()));
+        System.out.println("  _______________\n");
+    }
+
+    /**
+     * Marks a task as done.
+     *
+     * @param input The string representation of the index of the task to be marked.
+     */
+    public void markTask(String input) throws Exception {
+        int taskNum = Integer.parseInt(input) - 1;
+        Task task = getTask(taskNum);
+        task.markIsDone();
+    }
+
+    /**
+     * Unmark a task if it is done.
+     *
+     * @param input The string representation of the index of the task to be unmarked.
+     */
+    public void unmarkTask(String input) throws Exception {
+        int taskNum = Integer.parseInt(input) - 1;
+        Task task = getTask(taskNum);
+        task.unmarkIsDone();
+    }
+
+    public void deleteTask(String input) throws Exception {
+        int taskNum = Integer.parseInt(input) - 1;
+        Task task = getTask(taskNum);
+        try {
+            tasks.remove(taskNum);
+
+            System.out.println("\n  _______________");
+            System.out.println("  Removed: " + task);
+            System.out.println(String.format("  Now you have %d tasks in the list", tasks.size()));
+            System.out.println("  _______________\n");
+        } catch (Exception e) {
+            throw new DukeException(String.format("Task number %d not found", taskNum));
+        }
+    }
+
+    public Task getTask(int taskNum) throws Exception {
+        Task task;
+
```

Comment:
I think for leaving blank spaces part maybe can be more consistent (based on what i saw on CS2103 requirement haha) so since your code mainly does not have black space before the ```try```, for this you can remove the black space too at line 72.